### ISSUE TITLE
Support for using depend serverless.yml as modules in single project

### ISSUE THREAD

**Issue by nikita-clearscale** [serverless/serverless#7436]
# Feature Proposal
Add possibility to define:
1. root serverless.yml
1. depended serverless.yml which
   1. should run in order
   1. or in parallel if child serverless.yml are independ
1. child serverless should use code from root `node_modules`

## Description
The best practise is to split a lot of resoure from single serverless file to multiple ones. But in this case even if resources as split into diffferent file/stack they depends on each other. This leads to:
1. A maintainer have to remeber the deploy order because there is no way to define the order for depend serverless application.
2. For each servreless folder there is a need to run `npm install`, there are multiple `node_modules` folder with 90% same files -> this slow down IDE working and deploy.

# What is the use case that should be solved.
I am working with about 20+ serveless stacks. The structure like that:

1. downloaders applications downloads the data
1. converters applications converts the data
1. processor applications do some logic

So it is obviuos that each group depend on each other: `downloaders -> converters -> processor`, but application in same group can be deployed into parallel. For example `my-donwloader/serverless.yml` can deploy with `other-donwloader/serverless.yml`. While `my-converter/serverless.yml` must wait for `my-donwloader/serverless.yml`.

1. If there is additional config how would it look
It would nice to have:
   1. root serverless.yml with modules section where all modules are defined
   1. each serverless.yml defines modules from which it depends.
   1. before deploy serverless build dependecy tree and run deploy from top to bottom. Serverless stack on same level deployed in parallel.

As additional commands serverelss framework should have a command to show build tree.
Each build process should duplicate log output into `.serverless` folder (because parallel outputs to same console not comfortable)

# How it can be implemented
May be walk throuth folders from modules section, build tree and then spawn nodejs process with build commands.

**Comment by medikoo** [comment]
@nikita-clearscale thanks for suggestion. Still have you looked into Serverless Components?

Idea behond components is to modularize configuration of serverless stack, and I believe it serves well the use case you're after.

Serverless Components can also be considered a future version of Serverlsess Framework, although it's still in experimental stage.
